### PR TITLE
LEAN-3014: Move contributor_list & affiliation_list to the body of the document

### PR DESCRIPTION
**Pull request by mbetamony**

### CODE REVIEW

File: src/jats/importer/jats-body-dom-parser.ts [hunk 1]
```diff
@@ -594,6 +594,135 @@ const nodes: NodeRule[] = [
...
   },
+  {
+    tag: 'sec[sec-type="affiliations"]',
+    node: 'affiliations_section', // NOTE: higher priority than 'section'
+    getAttrs: (node) => {
+      const element = node as HTMLElement
+
+      return {
+        id: element.getAttribute('id'),
+        category: chooseSectionCategory(element),
```
Comment (mnatsheh) [inline src/jats/importer/jats-body-dom-parser.ts:605]: Can this be static?

File: src/jats/importer/jats-body-dom-parser.ts [hunk 2]
```diff
@@ -594,6 +594,135 @@ const nodes: NodeRule[] = [
...
+  },
+  {
+    tag: 'sec[sec-type="contributors"]',
+    node: 'contributors_section', // NOTE: higher priority than 'section'
+    getAttrs: (node) => {
+      const element = node as HTMLElement
+
+      return {
+        id: element.getAttribute('id'),
+        category: chooseSectionCategory(element),
```
Comment (mnatsheh) [inline src/jats/importer/jats-body-dom-parser.ts:664]: Can this be static?

**Review by mnatsheh — APPROVED** [review]
Be sure to coordinate this work with what Hashem is doing in [LEAN-3030](https://jira.atypon.com/browse/LEAN-3030).

File: src/jats/importer/jats-body-dom-parser.ts (hunk 2 above)
Comment (mbetamony) [reply src/jats/importer/jats-body-dom-parser.ts:664]: Do you mean something like this?
```
getAttrs: () => {
      return {
        id: 'MPSection:contributors',
        category: 'MPSectionCategory:contributors',
      }
    },
    
```

Comment (mbetamony) [reply src/jats/importer/jats-body-dom-parser.ts:664]: If we want to keep the `id` always to be `MPSection:contributors`, I'll have to exclude it from [jats-parser-utils.ts](https://github.com/Atypon-OpenSource/manuscripts-transform/pull/50/files/faaa1649a542d3b990ff8047420b2d76702133da#diff-dbc411c09874e29105b449f718f7e2337ce776db58e43a75fa8133edf67c1630R79), and make it static in [decoder.ts](https://github.com/Atypon-OpenSource/manuscripts-transform/pull/50/files/faaa1649a542d3b990ff8047420b2d76702133da#diff-c89cc32a6cfe692e9c8fb06947554b63975e88b61d2ce89074fb17d891d483a1R866).
And the same thing for the affiliations section. 
That should be easy to do.

Comment (mnatsheh) [reply src/jats/importer/jats-body-dom-parser.ts:664]: We don't need the id to be static. I was talking about the category.

Comment (mbetamony) [reply src/jats/importer/jats-body-dom-parser.ts]: Done

File: src/jats/importer/jats-body-transformations.ts [hunk 3]
```diff
@@ -401,6 +406,83 @@ export const jatsBodyTransformations = {
...
+        affiliation.addressLine1 &&
+          item.setAttribute('addressLine1', affiliation.addressLine1)
+        affiliation.addressLine2 &&
+          item.setAttribute('addressLine2', affiliation.addressLine2)
+        affiliation.addressLine3 &&
+          item.setAttribute('addressLine3', affiliation.addressLine3)
+        affiliation.postCode &&
+          item.setAttribute('postCode', affiliation.postCode)
+        affiliation.country && item.setAttribute('country', affiliation.country)
+        ;(affiliation.priority == 0 || affiliation.priority) &&
```
Comment (mbetamony) [inline src/jats/importer/jats-body-transformations.ts:436]: @mnatsheh FYI after your review, I added priority to each affiliations & contributors model, as manuscripts-api failed to validate these models.
(I also had to add them to the encoder/decoder/jats-body-parser)

**Comment by mbetamony** [comment]
I've tried deploying the new transformer, but I'm getting this error on the dashboard. Do you guys know why this would happen? Do I need to change anything in the html export or toDom in the contributor/contributor_section nodes, and affiliation/affiliation_section nodes?
@mbartenev-atypon @mnatsheh 
image

**Pull request merged**